model_access_restriction/tests/test_model_access_restriction.py

```diff
@@ -0,0 +1,48 @@
+# Copyright 2025 Alberto Martínez <alberto.martinez@sygel.es>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo.tests.common import TransactionCase
+
+
+class TestModelAccessRestriction(TransactionCase):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.model = cls.env.ref("base.model_res_partner")
+        cls.group = cls.env["res.groups"].create(
+            {
+                "name": "Test Group",
+            }
+        )
+        cls.restriction = cls.env["ir.model.access.restriction"].create(
+            {
+                "name": "Test Restriction",
+                "model_id": cls.model.id,
+                "perm_create": True,
+                "perm_unlink": True,
+                "groups": [(4, cls.group.id)],
+            }
+        )
+
+    def test_one_restriction(self):
```

Comment:
I think it would be more clear using with_user and assertRaises with AccessError

Reply:
Do you mean changing all the tests to try to create instead calling check_access_rights with create?

Reply:
Exactly, but it's just a suggestion.

Reply:
OK